src/Microsoft.ML.FastTree/FastTreeArguments.cs

```diff
@@ -442,6 +442,17 @@ internal virtual void Check(IExceptionContext ectx)
         }
     }
 
+    /// <summary>
+    /// Early stopping metrics for Regression and Ranking tasks in <see cref="FastTree"/>
+    /// </summary>
+    public enum EarlyStoppingMetrics
+    {
+        RegressionL1 = 1,
+        RegressionL2 = 2,
+        RankingNdcg1 = 1,
+        RankingNdcg3 = 3
```

Comment:
Please have separate enums for each task. The argument subclasses for each task will have as a public property something that takes that as its input. The existing argument can be then made internal. I am also pretty curious what the state is for binary classification.

Note that given the current state of the code, assigning `RankingNdcg1` to a regression task is totally legal, and it will happen to just map to L1, which is surely nonsense. The goal of #2521 is about introducing an enum, it is sure, but the udnerlying point really is to have an API that makes some degree of sense.

Reply:
Ditto what Tom said. I'd like the API to be like `TaskEarlyStoppingMetrics.Metric`.

Also about Binary Classification: When I was playing around with the API, the early stopping for Binary Classification wasn't clear. It is possible we have hard-coded the stopping metric we use.